### ISSUE TITLE
Gradient

### ISSUE THREAD

**Issue by HugoRodriguesQW** [HugoRodriguesQW/neuron#4]
## Gradients
>Gradient, specifically gradient descent, is an optimization algorithm used to adjust neural network weights during training. The objective of the training is to minimize a loss function, which measures the difference between the outputs programmed by the network and the actual outputs of the training set. The gradient is derived from the loss function with respect to the network weights, and it points in the direction in which the loss function is serviced most quickly.
>
>In the training process, the gradient is calculated in relation to the network weights and used to update these weights in order to minimize the loss function. The gradient determines the magnitude and direction of updating the weights in each training iteration. This allows the network to adjust to the training data and learn to perform a specific task for the quality being trained.
> -by chatGPT

## Gradient Descent (GD) and  Stochastic Gradient Descent:
<img width="660" alt="opt_comparison" src="https://github.com/HugoRodriguesQW/neuron/assets/71078903/77772c2a-b605-45ed-b8b5-50fa077c195c">

## Mini-batch
<img width="660" alt="opt_part" src="https://github.com/HugoRodriguesQW/neuron/assets/71078903/50d487f1-99db-4676-8d7e-d6a27c7276e4">

## Momentum
<img width="660" alt="opt_momentum" src="https://github.com/HugoRodriguesQW/neuron/assets/71078903/807a261f-e179-4893-a0f7-f402aa4329eb">

## Adam
<img width="660" alt="opt_momentum"  src="https://github.com/HugoRodriguesQW/neuron/assets/71078903/7e479d26-088f-4325-8b44-68b0e4da1bd7">

 ## RMSprop

<img width="660" alt="opt_momentum"  src="https://github.com/HugoRodriguesQW/neuron/assets/71078903/428be065-7339-410e-b26b-e01f23b3ea13">
